### DISCUSSION TITLE
Pathfinding mob has the zoomies

### DISCUSSION THREAD

**Discussion by ZakShearman** [Minestom/Minestom#1434] [Help] [answered]
Title says it all: https://youtu.be/kuoLgh0v2TQ

```java
    public WanderingMobExperiment(LobbyExtension extension) {
        super(EntityType.TURTLE);
        Instance instance = extension.getLobbyInstance();
        this.setInstance(instance, new Pos(3.5, 54, -11.5));

        this.addAIGroup(
                new EntityAIGroupBuilder()
                        .addGoalSelector(new RandomStrollGoal(this, 5))
                        .build()
        );
    }
```

How can I make the mob move at a reasonable speed and also, is it possible with the built in PF for it to not get caught on blocks?

**Comment by mworzala** [comment, marked as answer]
The builtin pathfinder has pretty [strange behavior](https://discord.com/channels/706185253441634317/706185253974442006/991667850722934935) ([reported](https://github.com/MadMartian/hydrazine-path-finding/issues/4) but the project seems abandoned).

As for the walking speed, you can set the [movement speed attribute](https://javadoc.minestom.net/net/minestom/server/attribute/Attribute.html#MOVEMENT_SPEED) on the entity and it will lower the speed.

**Reply by ZakShearman** [reply]
Ahh ok thx. I think I'll just wait until there's a good alternative and abandon this idea for now <3
